server/controllers/finance/reports/balance/index.js

```diff
@@ -0,0 +1,225 @@
+ */
+function processAccounts(balances, accounts, totals) {
+
+  // format and process opening balance for accounts
+  accounts = balances.beginning.reduce(function (init, row) {
+    var account = init, id = row.number;
+    var obj = account[id] = {}, sold = getSold(row);
+    obj.label = row.label;
+    obj.beginDebit = sold.debit;
+    obj.beginCredit = sold.credit;
+    obj.middleDebit = 0;
+    obj.middleCredit = 0;
+    obj.is_charge = row.is_charge;
+    obj.is_asset = row.is_asset;
+    account[id] = obj;
+    return account;
+  }, {});
+
+  // format and process the monthly balance for accounts
+  balances.middle.forEach(function (row){
+    var account = accounts[row.number] || {};
+    account.middleDebit = row.debit;
+    account.middleCredit = row.credit;
+    account.label = row.label;
+    account.is_charge = row.is_charge;
+    account.is_asset = row.is_asset;
+    accounts[row.number] = account;
+  });
+
+  Object.keys(accounts).forEach(function (item){
+    accounts[item].endDebit = 0;
+    accounts[item].endCredit = 0;
+    var sold = (accounts[item].beginDebit || 0 - accounts[item].beginCredit || 0) + (accounts[item].middleDebit - accounts[item].middleCredit);
+    if(sold < 0){
+      accounts[item].endCredit = sold * -1;
+    }else{
+     accounts[item].endDebit = sold;
+    }
+  });
+
+  // process for getting totals
+  totals = Object.keys(accounts).reduce(function (totals, key) {
+    var account = accounts[key];
+    totals.beginDebit += (account.beginDebit || 0);
+    totals.beginCredit += (account.beginCredit || 0);
+    totals.middleDebit += (account.middleDebit || 0);
+    totals.middleCredit += (account.middleCredit || 0);
+    totals.endDebit += (account.endDebit || 0);
+    totals.endCredit += (account.endCredit || 0);
+    return totals;
+  }, {
+    beginDebit : 0,
+    beginCredit : 0,
+    middleDebit: 0,
+    middleCredit : 0,
+    endDebit : 0,
+    endCredit : 0
+  });
+
+  return { accounts, totals };
+}
+
+/**
+ * @function getSold
+ * @description return the sold of an account
+ * @param {object} account An objec
+ */
+function getSold (item){
+  var result = {debit : 0, credit : 0}, sold = 0;
+
+  if(item.is_asset === 1 || item.is_charge === 1){
+    sold = item.debit - item.credit;
+    if(sold < 0){
+      result.credit = sold * -1 ;
+    }else{
+      result.debit = sold;
+    }
+  }else{
+    sold = item.credit - item.debit;
+    if(sold < 0){
+      result.debit = sold * -1;
+    } else{
+      result.credit = sold;
+    }
+  }
+  return result;
+}
+
+/**
+ * @function balanceReporting
+ * @description return the balance needed according the given params
+ * @param {object} params An object which contains dates range and the account class
+ */
+function balanceReporting(params) {
+  let sql, hasClasse, periodPlage, queryParameters,
+      query = params,
+      data = {};
+
+  hasClasse = (query.classe !== '*');
+  periodPlage = (query.dateFrom && query.dateTo);
```

Comment:
`Plage` is `range` in English.  It would be best to keep the code all in the same language!